text/0000-deprecate-component-manager-string-lookup.md

```diff
@@ -0,0 +1,90 @@
+- Start Date: 2018-10-23
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This deprecates the string based lookup API for associating a custom component manager with a corresponding base class.
+
+```js
+import EmberObject from '@ember/object';
+import { setComponentManager } from '@ember/component';
+
+export default setComponentManager('basic', EmberObject.extend({
+  //...
+}))
+```
+
+Instead you must pass a factory function that produces a instance of the custom manager:
+
+```js
+import EmberObject from '@ember/object';
+import { createManager } from './basic-manager';
+import { setComponentManager } from '@ember/modifier';
+
+export default setComponentManager(createManager, EmberObject.extend({
+  // ...
+}));
+```
+
+Where `createManager` is:
+
+```js
+export function createManager(owner) {
+  return new BasicManager(owner);
+}
+```
+
+# Motivation
+
+There are several motivators:
+
+- A string based API is not friendly when it comes to tree shaking. Would force us into creating a compiler to turn the string into a symbol that build tools like Rollup and Webpack could analyze.
```

Comment:
```suggestion
- A string-based API is not friendly when it comes to tree shaking. It would force us into creating a compiler to turn the string into a symbol that build tools like Rollup and Webpack could analyze.
```